src/tracksdata/functional/_apply.py

```diff
@@ -0,0 +1,238 @@
+        The slicing of the tile with the overlap.
+    slicing_wo_overlap : tuple[slice, ...]
+        The slicing of the tile WITHOUT overlap.
+    """
+
+    graph_filter: BaseFilter
+    graph_filter_wo_overlap: BaseFilter
+
+    slicing: tuple[slice, ...]
+    slicing_wo_overlap: tuple[slice, ...]
+
+
+T = TypeVar("T")  # per tile return type
+R = TypeVar("R")  # reduce return type
+
+MapFunc = Callable[[Tile], T]
+ReduceFunc = Callable[[list[T]], R]
+
+
+@dataclass
+class TilingScheme:
+    """
+    Tiling scheme for the graph.
+    Graph will be sliced with 'tile' + 2 * 'overlap' per axis.
+
+    Parameters
+    ----------
+    tile : tuple[int, ...]
+        The shape of the tile.
+    overlap : tuple[int, ...]
+        The overlap between tiles.
+    attrs : list[str] | None, optional
+        The attributes to include in the tile. If None, all attributes will be included.
+        By default "t", "z", "y", "x" are included. If some columns are not present, they will be ignored.
+    """
+
+    tile: tuple[int, ...]
+    overlap: tuple[int, ...]
+    attrs: list[str] | None = None
+
+    def __post_init__(self) -> None:
+        if len(self.tile) != len(self.overlap):
+            raise ValueError(
+                "'TilingScheme.tile' and 'TilingScheme.overlap' must have the same length, "
+                f"got {len(self.tile)} and {len(self.overlap)}"
+            )
+
+        if any(tile <= 0 for tile in self.tile):
+            raise ValueError(f"'TilingScheme.tile' must be greater than 0, got {self.tile}")
+
+        if any(overlap < 0 for overlap in self.overlap):
+            raise ValueError(f"'TilingScheme.overlap' must be non-negative, got {self.overlap}")
+
+        if self.attrs is not None:
+            if len(self.attrs) != len(self.tile):
+                raise ValueError(
+                    f"'TilingScheme.attrs' must have the same length as 'TilingScheme.tile', "
+                    f"got {len(self.attrs)} and {len(self.tile)}"
+                )
+
+
+@overload
+def apply_tiled(
+    graph: BaseGraph,
+    tiling_scheme: TilingScheme,
+    func: MapFunc,
+    *,
+    agg_func: None,
+) -> Iterator[T]: ...
+
+
+@overload
+def apply_tiled(
+    graph: BaseGraph,
+    tiling_scheme: TilingScheme,
+    func: MapFunc,
+    *,
+    agg_func: ReduceFunc,
+) -> R: ...
+
+
+def _get_tiles_corner(
+    start: Sequence[int],
+    end: Sequence[int],
+    tiling_scheme: TilingScheme,
+) -> list[tuple[int, ...]]:
+    """
+    Get the corner of the tiles.
+
+    Parameters
+    ----------
+    start : Sequence[int]
+        The start of the graph.
+    end : Sequence[int]
+        The end of the graph.
+    tiling_scheme : TilingScheme
+        The tiling scheme to use.
+    """
+    eps = 1e-8  # adding eps because np.arange is right exclusive
+    tiles_corner = list(
+        itertools.product(
+            *[np.arange(s, e + eps, t).tolist() for s, e, t in zip(start, end, tiling_scheme.tile, strict=True)]
```

Comment:
Since start and end are ints, maybe we can do something like:
It may also be useful to note whether indices are inclusive or exclusive.

```suggestion
            *[np.arange(s, e+1, t).tolist() for s, e, t in zip(start, end, tiling_scheme.tile, strict=True)]
```